### PR TITLE
Add documentation for preview module

### PR DESCRIPTION
**Pull request by Alexsandruss**
## Description

Documentation for preview module and functionality.

How it is rendered:

image

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

N/A

### CODE REVIEW

File: doc/sources/preview.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+     - Is patching supported
+   * - EmpiricalCovariance
+     - sklearnex.preview.covariance
+     - Yes
+   * - IncrementalPCA
+     - sklearnex.preview.decomposition
+     - Yes
+   * - Ridge
+     - sklearnex.preview.linear_model
+     - Yes
```
Comment (samir-nasibli) [inline doc/sources/preview.rst:76]: Could you please add  docstrings for module data members or class attributes?
It could be done like here for non-scikit algos https://github.com/intel/scikit-learn-intelex/blob/main/doc/sources/non-scikit-algorithms.rst

Comment (samir-nasibli) [reply doc/sources/preview.rst:76]: Also does it make sense to add notes why exact algorithm added to this module?

Comment (Alexsandruss) [reply doc/sources/preview.rst:76]: > Could you please add docstrings for module data members or class attributes? It could be done like here for non-scikit algos https://github.com/intel/scikit-learn-intelex/blob/main/doc/sources/non-scikit-algorithms.rst

Why is full docstring for a class required here? It's just a list of estimators.

Comment (Alexsandruss) [reply doc/sources/preview.rst:76]: > Also does it make sense to add notes why exact algorithm added to this module?

Reason is the same - partial compliance with sklearn, as I remember. But it's hard to fully backtrack exact reasons so it might be skipped for now.

**Review by icfaust — APPROVED** [review]
